rotkehlchen/chain/evm/decoding/extrafi/decoder.py

```diff
@@ -0,0 +1,350 @@
+                event.counterparty = CPT_EXTRAFI
+                event.notes = f'Claim {event.balance.amount} {token.symbol_or_name()} from Extrafi'
+                break
+        else:
+            log.error(f'Could not find extrafi reward transfer at {context.transaction}')
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _vote_escrow_contract(self, context: DecoderContext) -> DecodingOutput:
+        """Handle lock of extra tokens in the protocol"""
+        if context.tx_log.topics[0] != USER_CHECKPOINT:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[0:32]),
+            token_decimals=18,
+        )
+
+        for event in context.decoded_events:
+            if (
+                event.asset == self.extra_token_id and
+                event.balance.amount == amount and
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == hex_or_bytes_to_address(context.tx_log.topics[2])
+            ):
+                locktime = deserialize_timestamp(hex_or_bytes_to_int(context.tx_log.topics[3]))
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_EXTRAFI
+                event.notes = f'Lock {amount} EXTRA until {timestamp_to_date(locktime, formatstr="%d/%m/%Y %H:%M:%S")}'  # noqa: E501
+                event.product = EvmProduct.STAKING
+                break
+        else:
+            log.error(f'Failed to find lock of EXTRA in {context.transaction}')
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _partially_close_farm_position(self, context: DecoderContext) -> DecodingOutput:
+        """Close a position in a farm partially"""
+        manager = hex_or_bytes_to_address(context.tx_log.topics[3])
+        amount_0_received = hex_or_bytes_to_int(context.tx_log.data[32:64])
+        amount_1_received = hex_or_bytes_to_int(context.tx_log.data[64:96])
+
+        for event in context.decoded_events:
+            if event.location_label != manager:
+                continue
+
+            if (
+                event.maybe_get_direction() == EventDirection.IN and
+                (
+                    asset_normalized_value(amount_0_received, event.asset) == event.balance.amount or  # type: ignore  # noqa: E501
+                    asset_normalized_value(amount_1_received, event.asset) == event.balance.amount  # type: ignore
+                )
+            ):
+                event.counterparty = CPT_EXTRAFI
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f"Withdraw {event.balance.amount} {event.asset.symbol_or_name()} from Extrafi's farm"  # noqa: E501
```

Comment:
Do farms have names? Or is it only one farm? If so shouldn't you either use the name or say: "from an Extrafi farm" ?